modules/civiremote_event/src/Controller/RegisterFormController.php

```diff
@@ -16,32 +16,114 @@
 namespace Drupal\civiremote_event\Controller;
 
 use Drupal;
+use Drupal\civiremote_event\CiviMRF;
 use Drupal\civiremote_event\Form\RegisterForm\RegisterFormInterface;
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Access\AccessResultAllowed;
+use Drupal\Core\Access\AccessResultNeutral;
 use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\Core\Routing\RouteMatch;
 use stdClass;
+use \Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
 
 class RegisterFormController extends ControllerBase {
 
-  public function form(stdClass $event = NULL, $profile = NULL) {
-    // Use default profile if not given.
-    if (!isset($profile) && isset($event)) {
-      $profile = $event->default_profile;
+  /**
+   * @var CiviMRF $cmrf
+   *   The CiviMRF service.
+   */
+  protected $cmrf;
+
+  /**
+   * @param CiviMRF $cmrf
+   *   The CiviMRF service.
+   */
+  public function __construct(CiviMRF $cmrf) {
+    $this->cmrf = $cmrf;
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   The Drupal service container.
+   *
+   * @return static
+   */
+  public static function create(ContainerInterface $container) {
+    /**
+     * Inject dependencies.
+     * @var CiviMRF $cmrf
+     */
+    $cmrf = $container->get('civiremote_event.cmrf');
+    return new static(
+      $cmrf
+    );
+  }
+
+  public function form(RouteMatch $route_match, string $context, stdClass $event = NULL, string $profile = NULL, stdClass $event_token = NULL) {
+    $event ??= $event_token;
+    $raw_event_token = $route_match->getRawParameter('event_token');
+
+    // Retrieve the form definition.
+    try {
+      $form = $this->cmrf->getForm(
+        (isset($event) ? $event->id : NULL),
+        $profile,
+        $raw_event_token,
+        $context
+      );
+      $fields = $form['values'];
+      $messages = $form['status_messages'] ?? [];
+      if (!empty($fields['profile']['value'])) {
+        $profile = $fields['profile']['value'];
+      }
+      else {
+        switch ($context) {
+          case 'create':
+            $profile = $event->default_profile;
+            break;
+          case 'update':
+            $profile = $event->default_update_profile;
+            break;
+          case 'cancel':
+            $profile = NULL;
+            break;
+          default:
+            throw new NotFoundHttpException(
+              $this->t('No profile found for CiviRemote event form.')
```

Comment:
IIRC the message of a `NotFoundHttpException` is not displayed to end users, so translation wouldn't be necessary.